packages/flutter_tools/gradle/src/main/kotlin/dependency_version_checker.gradle.kts

```diff
@@ -0,0 +1,270 @@
+                    )
+                )
+            }
+            else if (version < warnGradleVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "Gradle",
+                        version.toString(),
+                        warnGradleVersion.toString()
+                    )
+                )
+            }
+        }
+
+        fun checkJavaVersion(version : JavaVersion, project : Project) {
+            if (version < errorJavaVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "Java",
+                        version.toString(),
+                        errorJavaVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnJavaVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "Java",
+                        version.toString(),
+                        warnJavaVersion.toString()
+                    )
+                )
+            }
+        }
+
+        fun checkAGPVersion(version : Version, project : Project) {
+            if (version < errorAGPVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "AGP",
+                        version.toString(),
+                        errorAGPVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnAGPVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "AGP",
+                        version.toString(),
+                        warnAGPVersion.toString()
+                    )
+                )
+            }
+        }
+
+        fun checkKGPVersion(version : Version, project : Project) {
+            if (version < errorKGPVersion) {
+                throw GradleException(
+                    getErrorMessage(
+                        "Kotlin",
+                        version.toString(),
+                        errorKGPVersion.toString()
+                    )
+                )
+            }
+            else if (version < warnKGPVersion) {
+                project.logger.error(
+                    getWarnMessage(
+                        "Kotlin",
+                        version.toString(),
+                        warnKGPVersion.toString()
+                    )
+                )
+            }
+        }
+    }
+}
+
+
+// Helper class to parse the versions that are provided as plain strings (Gradle, Kotlin) and
+// perform easy comparisons.
+class Version(val major : Int, val minor : Int, val patch : Int) : Comparable<Version> {
+    companion object {
+        fun fromString(version : String) : Version {
+            val asList : List<String> = version.split(".")
+            return Version(
+                major = asList.getOrElse(0, {"0"}).toInt(),
```

Comment:
This will crash if you ever pass something that cant be converted to an int. Consider stronger input protections. For example `fromString('7.3.0-rc2')` should probably not crash.

Reply:
I added trimming to the gradle getting, and also changed the parsing so we won't crash. Also added a comment to the Gradle getting explaining the trimming, and more doc explanation on the Version class.